pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -336,6 +334,18 @@ func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Re
 		return reconcile.Result{}, err
 	}
 
+	// Recover Pods with updated init containers in case of fails
```

Comment:
@rigazilla , @ryanemerson I've added specific recovery logic from the failed pods in case of error happens during downloading or incorrect hash sum.